confluence-syntax-xhtml/src/main/java/org/xwiki/contrib/confluence/parser/xhtml/internal/ConfluenceXHTMLParser.java

```diff
@@ -124,6 +126,10 @@ public class ConfluenceXHTMLParser extends AbstractWikiModelParser
     @Named("plain/1.0")
     private StreamParser plainParser;
 
+    @Inject
+    @Named("xwiki/2.1")
```

Comment:
This is not accurate, as the content you plan to parse might not be xwiki/2.1 at all. It's decided by either `ConfluenceXHTMLInputProperties#getMacroContentSyntax()` or (if not set) the `RenderingContext#getTargetSyntax()`. And it's even possible no parser exist at all for this configured syntax (since it's only expected to be used for rendering  right now).

Now you don't really need to produce any string here as the caption is internal to the parser, so you could use the [same trick as XWiki 14.10+ for link labels](https://github.com/xwiki/xwiki-rendering/blob/master/xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiCommentHandler.java#L199): instead of serializing the caption as a string, you could serialize it as a XDOM, that way you don't need to worry about the syntax it's in when you are in `ConfluenceImageWikiReference#onImage`.